### PR TITLE
Refactor auto-backup in flashing tab for PWA

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

File: src/js/utils/AutoBackup.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        this.sendSerial(`${line}\n`, callback);
+    }
+
+    execute(callback) {
+        this.callback = callback;
+
+        const port = PortHandler.portPicker.selectedPort;
+
+        if (port.startsWith('serial')) {
+            const baud = parseInt($('#flash_manual_baud_rate').val()) || 115200;
```
Comment (McGiverGim) [inline src/js/utils/AutoBackup.js:128]: This value is at the PortHandler too, not need to read it with jquery.

File: src/js/utils/AutoBackup.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            gui_log(i18n.getMessage('serialPortOpenFail'));
+        }
+    }
+
+    handleDisconnect(event) {
+        if (event.detail) {
+            gui_log(i18n.getMessage('serialPortClosedOk'));
+        } else {
+            gui_log(i18n.getMessage('serialPortClosedFail'));
+        }
```
Comment (McGiverGim) [inline src/js/utils/AutoBackup.js:36]: Maybe simplify it?
```suggestion
        gui_log(i18n.getMessage(event.detail ? 'serialPortClosedOk' , 'serialPortClosedFail'));
```

**Review by McGiverGim — CHANGES_REQUESTED** [review]
I like to see refactors :)

Having timeouts it's not the best option, but I suppose it's the easiest at this point. Maybe we can try to remove them in a future refactor.

File: src/js/utils/AutoBackup.js (hunk 1 above)
Comment (haslinghuis) [reply src/js/utils/AutoBackup.js:128]: `flash_manual_baud_rate` is not the same as `PortHandler.portPicker.selectedBauds`

image

File: src/js/utils/AutoBackup.js (hunk 2 above)
Comment (haslinghuis) [reply src/js/utils/AutoBackup.js]: Done.

File: src/js/utils/AutoBackup.js (hunk 1 above)
Comment (McGiverGim) [reply src/js/utils/AutoBackup.js:125]: True, I didn't remember it. The backup must use this flash baud rate in place of the "connection" one?

Comment (haslinghuis) [reply src/js/utils/AutoBackup.js]: Updated to use the portPicker baud selection.

**Review by McGiverGim — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]

**Comment by nerdCopter** [comment]
`864305e6`, `yarn dev`
- the flashing's `Create Backup` button worked.

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**